parser-typechecker/src/Unison/CommandLine/InputPatterns.hs

```diff
@@ -709,49 +711,60 @@ resetRoot = InputPattern "reset-root" [] [(Required, pathArg)]
      pure $ Input.ResetRootI src
     _ -> Left (I.help resetRoot))
 
+pullSilent :: InputPattern
+pullSilent =
+  pullImpl "pull.silent" Verbosity.Silent
+
 pull :: InputPattern
-pull = InputPattern
-  "pull"
-  []
-  [(Optional, gitUrlArg), (Optional, pathArg)]
-  (P.lines
-    [ P.wrap
-      "The `pull` command merges a remote namespace into a local namespace."
-    , ""
-    , P.wrapColumn2
-      [ ( "`pull remote local`"
-        , "merges the remote namespace `remote`"
-        <>"into the local namespace `local`."
-        )
-      , ( "`pull remote`"
-        , "merges the remote namespace `remote`"
-        <>"into the current namespace")
-      , ( "`pull`"
-        , "merges the remote namespace configured in `.unisonConfig`"
-        <> "with the key `GitUrl.ns` where `ns` is the current namespace,"
-        <> "into the current namespace")
-      ]
-    , ""
-    , P.wrap "where `remote` is a git repository, optionally followed by `:`"
-    <> "and an absolute remote path, such as:"
-    , P.indentN 2 . P.lines $
-      [P.backticked "https://github.com/org/repo"
-      ,P.backticked "https://github.com/org/repo:.some.remote.path"
-      ]
-    ]
-  )
-  (\case
-    []    ->
-      Right $ Input.PullRemoteBranchI Nothing Path.relativeEmpty' SyncMode.ShortCircuit
-    [url] -> do
-      ns <- parseUri "url" url
-      Right $ Input.PullRemoteBranchI (Just ns) Path.relativeEmpty' SyncMode.ShortCircuit
-    [url, path] -> do
-      ns <- parseUri "url" url
-      p <- first fromString $ Path.parsePath' path
-      Right $ Input.PullRemoteBranchI (Just ns) p SyncMode.ShortCircuit
-    _ -> Left (I.help pull)
-  )
+pull = pullImpl "pull" Verbosity.Default
+
+pullImpl :: [Char] -> Verbosity -> InputPattern
```

Comment:
Minor thing - I'd say `String` instead of `[Char]` here - `String` is just an alias for `[Char]` in Haskell.